src/convertToObject.js

```diff
@@ -6,7 +6,34 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (!sourceString.length) {
+    return {};
+  }
+
+  const noEmptyLines = sourceString
+    .split(';')
+    .map((line) => {
+      return line.trim();
+    })
+    .filter((item) => item.length > 0);
```

Comment:
[CODE KNOWLEDGE] - There's no need to make a copy before each operation. Methods like `filter` do not mutate the original array, so you can chain them directly without using `map` for trimming. You can combine `split`, `map`, and `filter` in a single chain.